plugins/aea-ledger-solana/aea_ledger_solana/solana.py

```diff
@@ -0,0 +1,1138 @@
+        program_id = PublicKey(contract_address)
+        idl = Idl.from_json(json.dumps(contract_interface["idl"]))
+        pg = Program(idl, program_id)
+
+        pg.provider.connection = self.api
+
+        if bytecode_path is not None:
+            # opening for [r]eading as [b]inary
+            in_file = open(bytecode_path, "rb")
+            bytecode = in_file.read()
+        else:
+            bytecode = None
+        return {"program": pg, "bytecode": bytecode}
+
+    def get_deploy_transaction(
+        self,
+        contract_interface: Dict[str, str],
+        deployer_address: Address,
+        raise_on_try: bool = False,
+        **kwargs: Any,
+    ) -> Optional[JSONLike]:
+        """
+        Get the transaction to deploy the smart contract.
+
+        :param contract_interface: the contract interface.
+        :param deployer_address: The address that will deploy the contract.
+        :param raise_on_try: whether the method will raise or log on error
+        :param kwargs: the keyword arguments.
+        :returns tx: the transaction dictionary.
+        """
+
+        """
+        if contract_interface["bytecode"] is None or contract_interface["program_keypair"] is None:
+            raise ValueError("Bytecode or program_keypair is required")
+
+        # check if solana cli is installed
+        result = subprocess.run(
+            ["solana --version"], capture_output=True, text=True, shell=True)
+        if result.stderr != "":
+            raise ValueError(result.stderr)
+
+        # save keys in uint8 array temp
+        value = struct.unpack('64B', payer_keypair.entity.secret_key)
+        uint8_array = array.array('B', value)
+        payer_uint8 = uint8_array.tolist()
+        temp_dir_payer = Path(tempfile.mkdtemp())
+        temp_file_payer = temp_dir_payer / "payer.json"
+        temp_file_payer.write_text(str(payer_uint8))
+
+        value = struct.unpack(
+            '64B', contract_interface["program_keypair"].entity.secret_key)
+        uint8_array = array.array('B', value)
+        program_uint8 = uint8_array.tolist()
+        temp_dir_program = Path(tempfile.mkdtemp())
+        temp_file_program = temp_dir_program / "program.json"
+        temp_file_program.write_text(str(program_uint8))
+
+        t = SolanaCrypto(temp_file_payer)
+        temp_dir_bytecode = Path(tempfile.mkdtemp())
+        temp_file_bytecode = temp_dir_bytecode / "bytecode.so"
+        temp_file_bytecode.write_bytes(contract_interface["bytecode"])
+
+        cmd = f'''solana program deploy --url {DEFAULT_ADDRESS} -v --keypair {str(temp_file_payer)} --program-id {str(temp_file_program)} {str(temp_file_bytecode)}'''
+
+        result = subprocess.run(
+            [cmd], capture_output=True, text=True, shell=True)
+
+        if result.stderr != "":
+            raise ValueError(result.stderr)
+
+        return result.stdout
+        """
+        raise NotImplementedError
+
```

Comment:
Please explain

Reply:
@angrybayblade this is commented out code from the external developer. 
at the moment I can not find the way to deploy solana contract without app being used.